dingo/gw/waveform_generator/waveform_generator.py

```diff
@@ -85,14 +108,12 @@ def __init__(
         self.f_ref = f_ref
         self.f_start = f_start
 
-        self.lal_params = None
-        if mode_list is not None:
-            self.lal_params = self.setup_mode_array(mode_list)
-
         self.transform = transform
         self._spin_conversion_phase = None
         self.spin_conversion_phase = spin_conversion_phase
 
+        self.new_interface = new_interface
```

Comment:
It is slightly awkward for the superclass `WaveformGenerator` to have to know about the new interface. I see that you do this so that you can know whether to call `_convert_parameters_to_lal_frame` or `_convert_paramters_new_interface`. However, this would lead to errors if we instantiate `WaveformGenerator` with `new_interface = True`.

These issues could be avoided by renaming `_convert_parameters_to_lal_frame` to `_convert_parameters`, and overriding this method in the subclass. That way the desired method will get called, depending on the calling class type.

Reply:
Yes, argument and attribute eliminated